recipe/core.py

```diff
@@ -145,6 +147,24 @@ def __init__(
 
         self.dynamic_extensions = dynamic_extensions
 
+    @classmethod
+    def from_config(cls, shelf, obj):
+        """Construct a Recipe from a plain Python dictionary."""
+        obj = normalize_schema(recipe_schema, obj)
+        filters = [
+            parse_condition(filter, shelf.Meta.select_from)
+            if isinstance(filter, dict)
+            else filter
+            for filter in obj.get('filters', [])
+        ]
+        return cls(
+            shelf=shelf,
+            metrics=obj.get('metrics'),
+            dimensions=obj.get('dimensions'),
+            filters=filters,
+            order_by=obj.get('order_by'),
+        )
```

Comment:
When I think about this, I think it's more extensible to do

```
recipe = cls(shelf=shelf)
for k, v in obj.items():
   if callable(getattr(recipe, k, None)):
       getattr(recipe, k)(v)
```

This allows use of extensions

Reply:
I looked into the extension stuff. It doesn't look like this suggestion actually changes whether extensions are used or not. Recipe.__init__ just takes the values and passes them on to `self` methods, so it's equivalent. However, your comment about extensions made me realize that Recipe.from_config was not accepting/passing through `**kwargs`, so it was impossible to configure extensions if you're loading from a config.

So I ended up simplifying the code in a different way: now the `obj` passed in is just validated and then passed in as keywords (after post-processing the `filters` as I already was doing). And now from_config also takes additional `**kwargs` to be added, so things like extension classes can be specified while loading a recipe from a config.